app/models/piece.rb

```diff
@@ -84,4 +86,11 @@ def on_board?(end_point)
 		return true
 	end
 
+	def do_move!(pos_x, pos_y)
```

Comment:
watch your tab spacing.  I think everyone struggles with this so it's kind of getting messy.